### PR TITLE
feat: stop using directories outside of OUT_DIR in build time

### PR DESCRIPTION
**Pull request by nicolasauler**
# Description
From cargo documentation (https://doc.rust-lang.org/cargo/reference/environment-variables.html):

> In general, build scripts should not modify any files outside of OUT_DIR. It may seem fine on the first blush, but it does cause problems when you use such crate as a dependency, because there’s an implicit invariant that sources in .cargo/registry should be immutable. cargo won’t allow such scripts when packaging.

This indeed is happening when I try to run nix related shenanigans on Bipa Server, due to this dependency not following this guideline.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -1,18 +1,17 @@
-const BUILD_DIR: &str = "src/gen";
```
Comment (guissalustiano) [inline build.rs:1]: You should remove code from this file roo

Comment (nicolasauler) [reply build.rs:1]: done: https://github.com/bipa-app/lnd-rs/pull/65/commits/4f2c308ca94ad947663de6eee0cfa4621564ba24

**Review by lorenzolfm — APPROVED** [review]

File: build.rs [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
-        .compile_protos(PROTOS, &[PROTOS_NS])
+        .out_dir(out_dir)
+        .compile_protos(
+            &[
+                "protos/invoices.proto",
+                "protos/lightning.proto",
+                "protos/router.proto",
+                "protos/chainkit.proto",
+            ],
+            &["protos"],
```
Comment (lsunsi) [inline build.rs:14]: quer aproveitar a renomear pra proto a pasta?

**Review by lsunsi — APPROVED** [review]

File: build.rs (hunk 2 above)
Comment (nicolasauler) [reply build.rs]: done: https://github.com/bipa-app/lnd-rs/pull/65/commits/76b5a8b2da96fa8e992a385f02b208ed850b0756

**Pull request merged**